### ISSUE TITLE
missing fiona.Env() in read_file()

### ISSUE THREAD

**Issue by vhoofk** [geopandas/geopandas#881]
It seems that `read_file()` is missing the `fiona.Env()` block that was recently added to `to_file()`.

Without this the `GDAL_DATA` environment variable is not set anymore since Fiona 1.8 as can be read in this issue thread:

[https://github.com/Toblerity/Fiona/issues/673](https://github.com/Toblerity/Fiona/issues/673)

> quote by @sgillies 
>
> Here's the deal: Fiona 1.7.13 patches os.environ to set GDAL_DATA and PROJ_LIB when we called fiona.drivers() whether directly or indirectly. This is smelly and I've changed it. Fiona 1.8 no longer patches os.environ. Now you need to run code that will use CRS objects within a with fiona.Env(): block

In our case this leads to the following exception and GDAL warnings (only relevant part shown):

```
venv35/lib/python3.5/site-packages/geopandas/geodataframe.py:443: in to_crs
    geom = df.geometry.to_crs(crs=crs, epsg=epsg)
venv35/lib/python3.5/site-packages/geopandas/geoseries.py:304: in to_crs
    proj_in = pyproj.Proj(self.crs, preserve_units=True)
/opt/rh/rh-python35/root/usr/lib64/python3.5/site-packages/pyproj/__init__.py:358: in __new__
    return _proj.Proj.__new__(self, projstring)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   RuntimeError: b'no arguments in initialization list'

_proj.pyx:84: RuntimeError


ERROR 4: Unable to open EPSG support file gcs.csv.  Try setting the GDAL_DATA environment variable to point to the directory containing EPSG csv files.
ERROR 6: No translation for an empty SRS to PROJ.4 format is known.```

**Comment by jorisvandenbossche** [comment]
@vhoofk Thanks a lot for the report!

I am going to try to do a new geopandas release as soon as possible after new year (now on holidays), so will include a fix for this. In the meantime, a PR to fix it is certainly already welcome!

**Comment by jorisvandenbossche** [comment]
Closed by #916
